backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/SonarLintBlameResult.java

```diff
@@ -32,23 +32,32 @@
 public class SonarLintBlameResult {
 
   private final BlameResult blameResult;
+  private final Path gitRepoRelativeProjectBaseDir;
 
-  public SonarLintBlameResult(BlameResult blameResult) {
+  public SonarLintBlameResult(BlameResult blameResult, Path gitRepoRelativeProjectBaseDir) {
     this.blameResult = blameResult;
+    this.gitRepoRelativeProjectBaseDir = gitRepoRelativeProjectBaseDir;
   }
 
   /**
-   * @param filePath A path relative to the Git repository root
+   * @param projectDirRelativeFilePath A path relative to the Git repository root
    * @param lineNumbers Line numbers for which to check the latest change date. Numbering starts from `1`!
    * @return The latest changed date or an empty optional if the date couldn't be determined or any of the lines is modified
    */
-  public Optional<Date> getLatestChangeDateForLinesInFile(Path filePath, Collection<Integer> lineNumbers) {
+  public Optional<Date> getLatestChangeDateForLinesInFile(Path projectDirRelativeFilePath, Collection<Integer> lineNumbers) {
     validateLineNumbersArgument(lineNumbers);
     var fileBlameByPath = blameResult.getFileBlameByPath();
-    var blameForFile = fileBlameByPath.get(FilenameUtils.separatorsToUnix(filePath.toString()));
-    if (blameForFile == null) {
+
+    var blameForFileOpt = Optional.of(projectDirRelativeFilePath.toString())
+      .map(gitRepoRelativeProjectBaseDir::resolve)
+      .map(Path::toString)
+      .map(FilenameUtils::separatorsToUnix)
+      .map(fileBlameByPath::get);
+
+    if (blameForFileOpt.isEmpty()) {
       return Optional.empty();
     }
+    var blameForFile = blameForFileOpt.get();
```

Comment:
Redundant check.
```
return blameForFileOpt
      .map(blameForFile -> {
      ... here you can put everything below
      })
```

Reply:
Yes, thanks. I changed it. The new version is even better